src/Castle.Windsor/Windsor/Installer/FromAssembly.cs

```diff
@@ -115,13 +115,21 @@ public static IWindsorInstaller InDirectory(AssemblyFilter filter, InstallerFact
 		///     <c>MyApp.SuperFeatures.dll</c>, <c>mscorlib.dll</c> and <c>ThirdPartyCompany.UberControls.dll</c> the <c>MyApp.exe</c> and <c>MyApp.SuperFeatures.dll</c> will be scanned for installers, and other
 		///     assemblies will be ignored.
 		/// </remarks>
+#if FEATURE_GETCALLINGASSEMBLY
```

Comment:
Thanks, I've added a few more comments to those changes.

With regards to the unit tests good point, however I still think it would be best to standardise both platforms rather than heaps of extra conditional compilation.

Below are the 9 methods that use `GetCallingAssembly`, if they are just a single line that passes through the assembly to the implementation then unit testing them is much less important. `AllTypes` is already obsolete so I wouldn't worry about a unit test for that one. If we could duplicate a basic unit test for `Classes.FromThisAssembly()` and `FromAssembly.This()` both compiled only for .NET Framework I'd be happy for the rest to use the one that requires an assembly argument.

- `AllTypes.FromThisAssembly()` (calls `Classes.FromAssembly(assembly)`)
- `Classes.FromAssemblyInThisApplication()`
- `Classes.FromThisAssembly()`
- `Types.FromAssemblyInThisApplication()`
- `Types.FromThisAssembly()`
- `FromAssembly.InThisApplication()`
- `FromAssembly.InThisApplication(installerFactory)`
- `FromAssembly.This()`
- `FromAssembly.This(installerFactory)`